add is_observation and is_prediction to metadata

equinor-ruaj opened the pull request.

daniel-sol commented:
Generally I think that we should encourage more the use of __getitem__ rather than introducing 100's of properties.. This is discussed, and there is a general consensus with the sudo team

equinor-ruaj commented:
> Generally I think that we should encourage more the use of **getitem** rather than introducing 100's of properties.. This is discussed, and there is a general consensus with the sudo team

removed the explicit `property` definitions and updated examples to use format `object["data"]["is_observation"]`
